projects/kubernetes/kubernetes/build/lib/version.sh

```diff
@@ -21,9 +28,10 @@ function build::version::create_env_file() {
     local -r repository="$4"
     local -r release_branch="$5"
 
-    git -C $repository checkout $tag
-    local -r source_date_epoch=$(git -C $repository show -s --format=format:%ct HEAD)
-
+    local -r source_date_epoch=$(git -C $repository show -s --format=format:%ct $tag^{commit})
```

Comment:
This checkout confused me for a while because I was trying to run the unit tests after running this during the build and they were passing, when they should fail due to a change in one of our patches.  Changed this up to avoid changing the branch so that if we ever do need to do anything with the kubernetes source its actually at the point that we built from.